Sources/PayPalNativeCheckout/PayPalUIInterfaces/NativeCheckout.swift

```diff
@@ -0,0 +1,26 @@
+
+import Foundation
+import UIKit
+@_implementationOnly import PayPalCheckout
+import PaymentsCore
+
+class NativeCheckout: CheckoutProtocol {
```

Comment:
Wondering if there is a better name for this class since it's just used for testing. Also if that's the case do we need it in our main interface or can we make this a mock class in our tests if that's it's main purpose? We should also add some docs on what this class is/why we need it above to make this clear.

Reply:
this is uded in `PaypalClient` it's not just for testing

Reply:
This is great - using a protocol here (`CheckoutProtocol`) is a great idea for us to be able to test `NativeCheckout.start()`!

Though, I agree with @jaxdesmarais that the naming is confusing, especially since `NativeCheckout` is a pretty loaded term/ has a lot of use throughout the SDK already. I think we can come up with better names for this class, along with the `CheckoutProtocol`.

---
Here are the [Swift API design guidelines](https://swift.org/documentation/api-design-guidelines/) on how to name protocols:

```
- Protocols that describe what something is should read as nouns (e.g. Collection).
- Protocols that describe a capability should be named using the suffixes `able`, `ible`, or `ing` (e.g. Equatable, ProgressReporting).
```

Reply:
Ideas? There already is a `PayPalClient.start()` so I wonder if we can think of something to remind us this class just basically wraps the 3rd party NXO framework 🤔 ...

`NativeCheckoutProvider: NativeCheckoutStartable`
`NativeCheckoutProvider: NativeCheckoutable`

Thoughts?

Reply:
I'm 1️⃣  for `NativeCheckoutStartable` it does fit the naming guidelines well.

Reply:
`NativeCheckoutProvider: NativeCheckoutStartable` this sounds good or `NativeCheckoutProvider: NativeCheckoutProtocol`

Reply:
@kgangineni - this still needs to be addressed in the code

Reply:
@kgangineni - Responding to your question in Slack DM here: the `NativeCheckoutProvider: NativeCheckoutStartable` is preferred since the later (`NativeCheckoutProvider: NativeCheckoutProtocol`) still uses `Protocol` in the name, which isn't in line with Swift style guide.